tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailRecoveryActionSetMethod.scala

```diff
@@ -152,12 +155,18 @@ class EmailRecoveryActionSetCreatePerformer @Inject()(val taskManager: TaskManag
         .left.map(errors => new IllegalArgumentException(ResponseSerializer.serialize(JsError(errors)).toString))
     } yield parsedRequest
 
-  private def submitTask(clientId: EmailRecoveryActionCreationId, userToRestore: Username, creationRequest: EmailRecoveryActionCreationRequest): SMono[CreationResult] =
-    SMono.fromCallable(() => taskManager.submit(new DeletedMessagesVaultRestoreTask(restoreService, userToRestore,
-      creationRequest.asQuery(configuration.maxEmailRecoveryPerRequest))))
+  private def submitTask(clientId: EmailRecoveryActionCreationId, userToRestore: Username, creationRequest: EmailRecoveryActionCreationRequest): SMono[CreationResult] = {
+    val fifteenDaysAgo = ZonedDateTime.now().minusDays(15)
+    val deletionDateLessThanFifteenDaysOldCriterion = CriterionFactory.deletionDate().afterOrEquals(fifteenDaysAgo)
+
+    val requestCriteria = creationRequest.asQuery(configuration.maxEmailRecoveryPerRequest).getCriteria
+    requestCriteria.add(deletionDateLessThanFifteenDaysOldCriterion)
```

Comment:
Work on mutable collection spotted.
Create a new collection instead and keep the previous collection unchanged.